apps/cowswap-frontend/src/legacy/hooks/useActivityDerivedState.ts

```diff
@@ -118,7 +118,14 @@ export function getActivityLinkUrl(params: {
   const { chainId, id, enhancedTransaction, order } = params
 
   if (enhancedTransaction) {
-    const { transactionHash, safeTransaction } = enhancedTransaction
+    const { transactionHash, hash, safeTransaction, hashType } = enhancedTransaction
+
+    /**
+     * This is a special case for Gnosis Safe transactions created via WC in a Safe with 1/1 signers
+     */
+    if (hashType === HashType.GNOSIS_SAFE_TX && hash) {
+      return getEtherscanLink(chainId, 'transaction', hash)
+    }
```

Comment:
Normal Safe transactions don't have `hash` property, they have `transactionHash` instead.
I know, this is super implicit but this is what we already have and I decided to not change it now.
Anyway, we 100% should refactor `EnhancedTransactionDetails` and everything related.

Reply:
At least I would add this implicit part as part of the comment because  the comment mentions only about the conclusion and not they reason that took you to it